python_src/src/lamp_py/migrations/versions/performance_manager_prod/sql_strings/strings_003.py

```diff
@@ -0,0 +1,89 @@
+                        static_calendar.friday AS friday,
+                        static_calendar.saturday AS saturday,
+                        static_calendar.sunday AS sunday
+                FROM static_calendar
+                WHERE static_calendar.static_version_key >=
+                    (SELECT min(static_calendar_dates.static_version_key) AS min_1
+                    FROM static_calendar_dates)) AS static_cal,
+                generate_series(static_cal.start_date, static_cal.end_date, '1 day') AS new_sd
+            WHERE (extract(dow
+                            from new_sd) = 0
+                    AND static_cal.sunday)
+                OR (extract(dow
+                            from new_sd) = 1
+                    AND static_cal.monday)
+                OR (extract(dow
+                            from new_sd) = 2
+                    AND static_cal.tuesday)
+                OR (extract(dow
+                            from new_sd) = 3
+                    AND static_cal.wednesday)
+                OR (extract(dow
+                            from new_sd) = 4
+                    AND static_cal.thursday)
+                OR (extract(dow
+                            from new_sd) = 5
+                    AND static_cal.friday)
+                OR (extract(dow
+                            from new_sd) = 6
+                    AND static_cal.saturday)) AS service_ids
+        UNION SELECT service_ids_special.service_id AS service_id,
+                        service_ids_special.service_date AS service_date,
+                        service_ids_special.static_version_key AS static_version_key
+        FROM
+            (SELECT static_calendar_dates.service_id AS service_id,
+                    static_calendar_dates.date AS service_date,
+                    static_calendar_dates.static_version_key AS static_version_key
+            FROM static_calendar_dates
+            WHERE static_calendar_dates.exception_type = 1) AS service_ids_special) AS all_service_ids
+        FULL OUTER JOIN
+        (SELECT static_calendar_dates.service_id AS service_id,
+                static_calendar_dates.date AS service_date,
+                static_calendar_dates.static_version_key AS static_version_key,
+                true AS to_exclude
+        FROM static_calendar_dates
+        WHERE static_calendar_dates.exception_type = 2) AS service_ids_exclude ON all_service_ids.service_id = service_ids_exclude.service_id
+        AND service_ids_exclude.service_date = all_service_ids.service_date
+        AND all_service_ids.static_version_key = service_ids_exclude.static_version_key
+        JOIN
+        (SELECT replace(mod_sd::date::text, '-', '')::integer AS service_date,
+                static_version_key
+        FROM
+            (SELECT feed_start_date::text::date AS start_date,
+                    lead(feed_start_date, 1, feed_end_date) OVER (ORDER BY feed_start_date)::text::date - 1 AS end_date,
```

Comment:
This is the only line that changed in the view, the default value for the `lead` function is now `feed_end_date`, instead of the current date.

Reply:
i don't quite follow what this change does?

Reply:
This subquery is responsible for creating a list of  all possible service_date's and their associated static_version_key, so essentially which static schedule applies to which service_date. Previously, that list creation stopped at the current date (when the query was executed) but the change makes to go until the feed_end_date of the last static schedule we have. 

If you have a lamp local_rds instance up and running (with some static data in it), the difference can be seen by comparing the results of these two queries: 
```
SELECT 
        TO_CHAR(mod_sd, 'YYYYMMDD')::integer AS service_date,
        static_version_key
FROM
(
        SELECT 
                feed_start_date::text::date AS start_date,
                lead(feed_start_date, 1, feed_end_date) OVER (ORDER BY feed_start_date)::text::date - 1 AS end_date,
                static_version_key
        FROM public.static_feed_info
) AS mod_fed_dates,
generate_series(mod_fed_dates.start_date, mod_fed_dates.end_date, '1 day') AS mod_sd
;
```

```
SELECT 
        TO_CHAR(mod_sd, 'YYYYMMDD')::integer AS service_date,
        static_version_key
FROM
(
        SELECT 
                feed_start_date::text::date AS start_date,
                lead(feed_start_date, 1, TO_CHAR(CURRENT_DATE, 'YYYYMMDD')::integer) OVER (ORDER BY feed_start_date)::text::date - 1 AS end_date,
                static_version_key
        FROM public.static_feed_info
) AS mod_fed_dates,
generate_series(mod_fed_dates.start_date, mod_fed_dates.end_date, '1 day') AS mod_sd
;
```